src/borg/repository.py

```diff
@@ -207,8 +228,13 @@ def check_can_create_repository(self, path):
             repository, user's can only use the quota'd repository, when their --restrict-to-path points
             at the user's repository.
         """
-        if os.path.exists(path) and (not os.path.isdir(path) or os.listdir(path)):
-            raise self.AlreadyExists(path)
+        if os.path.exists(path):
+            if not os.path.isdir(path):
+                raise self.NotADirectory(path)
+            elif self.is_repository(path):
+                raise self.AlreadyExists(path)
+            elif os.listdir(path):
+                raise self.OtherFilesExist(path)
```

Comment:
strictly taken, you don't need "ELif" if you raise in every block.

Reply:
👍 changed to plain `if`s